File: nim/main3.nim
proc dangerous(a: seq[int], index: int): int =
  result = a[index] # IndexError

proc foo(a: seq[int], counter: int): int =
  if counter == 0:
    result = dangerous(a, counter + 9137)
  else:
    result = foo(a, counter - 1)

when isMainModule:
  var a = newSeq[int](1000)
  let r = foo(a, 900)
  echo "The result is ", r

File: php/main3.php
<?php
function dangerous($array, $index) {
    return $array[$index] + 1; // will be null+1 -> TypeError in PHP 8
}
function foo($array, $counter) {
    if ($counter === 0) return dangerous($array, $counter + 9137);
    return foo($array, $counter - 1);
}
$array = array_fill(0, 1000, 0);
$result = foo($array, 900);
echo "The result is $result\n";
